test/extended/rorfs/readonly_root_filesystem.go

```diff
@@ -4,158 +4,112 @@ import (
-		var (
-			randmStr = uuid.New()
-			testNs1  = "test-rofs-cm-" + randmStr
-			testNs2  = "test-rofs-rm-" + randmStr
-		)
-		// Define namespaces to test based on platform availability
-		allNamespaces := []string{
-			"openshift-controller-manager-operator",
-			"openshift-route-controller-manager",
-			"openshift-cluster-version",
-			"openshift-image-registry",
-			"openshift-insights",
-			"openshift-dns",
-			"openshift-etcd",
-			"openshift-etcd-operator",
-			"openshift-kube-scheduler-operator",
-			"openshift-kube-scheduler",
-			"openshift-kube-apiserver-operator",
-			"openshift-kube-apiserver",
-			"openshift-cloud-credential-operator",
-			"openshift-marketplace",
-			"openshift-console-operator",
-			"openshift-console",
-			"openshift-operator-lifecycle-manager",
-			"hypershift",
-		}
+	if len(podList.Items) == 0 {
+		g.Skip(fmt.Sprintf("No suitable pods found in namespace %s", namespace))
+	}
 
-		// Filter namespaces that actually exist in the cluster using kube client
-		var namespaces []string
-		for _, ns := range allNamespaces {
-			_, err := oc.AdminKubeClient().CoreV1().Namespaces().Get(context.Background(), ns, metav1.GetOptions{})
-			if err == nil {
-				namespaces = append(namespaces, ns)
-				e2e.Logf("Namespace %s exists, will be tested", ns)
-			} else {
-				e2e.Logf("Namespace %s does not exist, skipping", ns)
+	// Filter out pods that should be skipped based on name patterns
+	var filteredPods []corev1.Pod
+	skipPatterns := []string{
+		"apiserver-watcher",
+	}
+
+	for _, pod := range podList.Items {
+		// Skip pods with specific name patterns
+		shouldSkip := false
+		for _, pattern := range skipPatterns {
+			if strings.Contains(strings.ToLower(pod.Name), strings.ToLower(pattern)) {
+				e2e.Logf("Skipping pod %s in namespace %s (matches exclusion pattern: %s)", pod.Name, namespace, pattern)
+				shouldSkip = true
+				break
 			}
 		}
-
-		if len(namespaces) == 0 {
-			g.Skip("No testable namespaces found in this cluster")
+		if shouldSkip {
+			continue
 		}
+		filteredPods = append(filteredPods, pod)
+	}
 
-		oc.SetupProject()
+	if len(filteredPods) == 0 {
+		g.Skip(fmt.Sprintf("No suitable pods found in namespace %s after filtering", namespace))
+	}
 
-		for _, ns := range namespaces {
-			// Get pods using optimized function with label selector to exclude unwanted pods
-			labelSelector := "app notin (guard,installer,pruner,revision)"
-			podNames, err := exutil.GetPodsListByLabel(oc.AsAdmin(), ns, labelSelector)
-			if err != nil {
-				e2e.Failf("Failed to get pods from namespace %s: %v", ns, err)
-			}
+	e2e.Logf("Found %d pods to test in namespace %s (after filtering)", len(filteredPods), namespace)
+	for _, pod := range filteredPods {
+		// Wait for pod to be ready before testing
+		e2e.Logf("Waiting for pod %s in namespace %s to be ready", pod.Name, namespace)
+		err := exutil.AssertPodToBeReady(oc, pod.Name, namespace)
```

Comment:
this should be removed, I assume it was a pre-check before trying to write to every container, but if that goes away per the comment below, the ready check can also go away. 

If we do decide to keep writing to every container (unlikely) I would just skip any container that is not ready. (using the data from your initial pod list above) No need for additional api requests and polling.

Reply:
@dgoodwin This is necessary sometimes pods are not ready and we are writing to it will fail the case. We should check end to end not only config, also check config is working correctly or not. otherwise we will regression.
i check now case timing only max 9 mins

Reply:
Earlier we see after bug fixes we didn't see, So I have update the test check only config.